src/subdomains/core/buy-crypto/process/entities/buy-crypto.entity.ts

```diff
@@ -204,9 +207,14 @@ export class BuyCrypto extends IEntity {
 
   calculateOutputReferenceAmount(price: Price): this {
     this.outputReferenceAmount = price.convert(this.inputReferenceAmountMinusFee, 8);
+    this.addPriceSteps(price?.steps);
     return this;
   }
```

Comment:
If inputAsset is not the same as inputReferenceAsset, add a price step (first one) with source Bank